plugins/checksum/checksum.c

```diff
@@ -22,244 +22,431 @@
 
 #include "config.h"
 
-#include <stdlib.h>
+#include <gio/gio.h>
 #include <glib.h>
+#include <glib/gerror.h>
 #include <glib/gstdio.h>
-#include <gio/gio.h>
+#include <stdlib.h>
 
 #include "hexchat-plugin.h"
 
 #define BUFSIZE 32768
-#define DEFAULT_LIMIT 256									/* default size is 256 MiB */
+#define DEFAULT_LIMIT 256 /* default size is 256 MiB */
 #define SHA256_DIGEST_LENGTH 32
 #define SHA256_BUFFER_LENGTH 65
 
-static hexchat_plugin *ph;									/* plugin handle */
+static hexchat_plugin *ph; /* plugin handle */
 static char name[] = "Checksum";
 static char desc[] = "Calculate checksum for DCC file transfers";
-static char version[] = "3.1";
-
-static void
-set_limit (char *size)
-{
-	int limit = atoi (size);
-
-	if (limit > 0 && limit < INT_MAX)
-	{
-		if (hexchat_pluginpref_set_int (ph, "limit", limit))
-			hexchat_printf (ph, "Checksum: File size limit has successfully been set to: %d MiB\n", limit);
-		else
-			hexchat_printf (ph, "Checksum: File access error while saving!\n");
-	}
-	else
-	{
-		hexchat_printf (ph, "Checksum: Invalid input!\n");
-	}
+static char version[] = "3.3";
+
+static void set_limit(char *size) {
+  int limit = atoi(size);
+
+  if (limit > 0 && limit < INT_MAX) {
+    if (hexchat_pluginpref_set_int(ph, "limit", limit))
+      hexchat_printf(
+          ph,
+          "Checksum: File size limit has successfully been set to: %d MiB\n",
+          limit);
+    else
+      hexchat_printf(ph, "Checksum: File access error while saving!\n");
+  } else {
+    hexchat_printf(ph, "Checksum: Invalid input!\n");
+  }
 }
 
-static int
-get_limit (void)
-{
-	int size = hexchat_pluginpref_get_int (ph, "limit");
+static int get_limit(void) {
+  int size = hexchat_pluginpref_get_int(ph, "limit");
 
-	if (size <= 0 || size >= INT_MAX)
-		return DEFAULT_LIMIT;
-	else
-		return size;
+  if (size <= 0 || size >= INT_MAX)
+    return DEFAULT_LIMIT;
+  else
+    return size;
 }
 
-static gboolean
-check_limit (GFile *file)
-{
-	GFileInfo *file_info;
-	goffset file_size;
+static gboolean check_limit(GFile *file) {
+  GFileInfo *file_info;
+  goffset file_size;
+
+  file_info = g_file_query_info(file, G_FILE_ATTRIBUTE_STANDARD_SIZE,
```

Comment:
This is a blocking IO call. There is an async version.

TBH I think it may make sense to use `GTask` and run the blocking IO in a thread with `g_task_run_in_thread()`. (Do be careful to not call `hexchat_` API in that thread).

Reply:
So should I apply these fixes, or reqriet it with the task, I personally think, the async read is better, so maybe run it async in a new thread, or would that be overkill ?

Reply:
No it doesn't make sense to combine the two.

The async way can be fine, its just more code thats harder to read.

Reply:
true